contracts/src/v0.8/automation/dev/MercuryRegistry.sol

```diff
@@ -0,0 +1,309 @@
+pragma solidity 0.8.6;
+
+import "../../shared/access/ConfirmedOwner.sol";
+import "../interfaces/AutomationCompatibleInterface.sol";
+import "../interfaces/StreamsLookupCompatibleInterface.sol";
+import "../../ChainSpecificUtil.sol";
+
+/*--------------------------------------------------------------------------------------------------------------------+
+| Mercury + Automation                                                                                                |
+| ________________                                                                                                    |
+| This implementation allows for an on-chain registry of price feed data to be maintained and updated by Automation   |
+| nodes. The upkeep provides the following advantages:                                                                |
+|   - Node operator savings. The single committee of automation nodes is able to update all price feed data using     |
+|     off-chain feed data.                                                                                            |
+|   - Fetch batches of price data. All price feed data is held on the same contract, so a contract that needs         |
+|     multiple sets of feed data can fetch them while paying for only one external call.                              |
+|   - Scalability. Feeds can be added or removed from the contract with a single contract call, and the number of     |
+|     feeds that the registry can store is unbounded.                                                                 |
+|                                                                                                                     |
+| Key Contracts:                                                                                                      |
+|   - `MercuryRegistry.sol` - stores price feed data and implements core logic.                                       |
+|   - `MercuryRegistryBatchUpkeep.sol` - enables batching for the registry.                                           |
+|   - `MercuryRegistry.t.sol` - contains foundry tests to demonstrate various flows.                                  |
+|                                                                                                                     |
+| TODO:                                                                                                               |
+|   - Optimize gas consumption.                                                                                       |
+-+---------------------------------------------------------------------------------------------------------------------*/
+contract MercuryRegistry is ConfirmedOwner, AutomationCompatibleInterface, StreamsLookupCompatibleInterface {
```

Comment:
can we add a comment somewhere to indicate that this only works with mercury v0.2? 

for mercury v0.3, certain feeds still have the same report schema but some are different (timestamp feeds)
and verify function will take different params..